### PR TITLE
Fix unnest rewriting logic

### PR DESCRIPTION
**Pull request by sgrebnov**
## 🗣 Description

Unnest unit tests were failing, there are some changes in Datafusion unnest logic that required some extra work:  improved unnest rewriting to correctly rewrite UnnestOptions (column names).

## 🔨 Related Issues

Fixes https://github.com/spiceai/spiceai/issues/3951

### CODE REVIEW

File: sources/sql/src/lib.rs [hunk 1]
```diff
@@ -1020,11 +1057,11 @@ mod tests {
         let tests = vec![
             (
                 "SELECT UNNEST([1, 2, 2, 5, NULL]), b, c from app_table where a > 10 order by b limit 10;",
-                r#"SELECT UNNEST(make_array(1, 2, 2, 5, NULL)), remote_table.b, remote_table.c FROM remote_table WHERE (remote_table.a > 10) ORDER BY remote_table.b ASC NULLS LAST LIMIT 10"#,
+                r#"SELECT UNNEST(make_array(1, 2, 2, 5, NULL)) AS "UNNEST(make_array(Int64(1),Int64(2),Int64(2),Int64(5),NULL))", remote_table.b, remote_table.c FROM remote_table WHERE (remote_table.a > 10) ORDER BY remote_table.b ASC NULLS LAST LIMIT 10"#,
```
Comment (sgrebnov) [inline sources/sql/src/lib.rs:1060]: This is unrelated to this change and related to datafusion behavior change when working with unnest.

File: sources/sql/src/lib.rs [hunk 2]
```diff
@@ -1020,11 +1057,11 @@ mod tests {
...
             ),
             (
                 "SELECT UNNEST(app_table.d), b, c from app_table where a > 10 order by b limit 10;",
-                r#"SELECT UNNEST(remote_table.d), remote_table.b, remote_table.c FROM remote_table WHERE (remote_table.a > 10) ORDER BY remote_table.b ASC NULLS LAST LIMIT 10"#,
+                r#"SELECT UNNEST(remote_table.d) AS "UNNEST(app_table.d)", remote_table.b, remote_table.c FROM remote_table WHERE (remote_table.a > 10) ORDER BY remote_table.b ASC NULLS LAST LIMIT 10"#,
```
Comment (sgrebnov) [inline sources/sql/src/lib.rs:1064]: Same as above

**Review by Sevenannn — APPROVED** [review]
